cpu/cortexm_common/include/cpu.h

```diff
@@ -167,6 +167,14 @@ static inline void cortexm_sleep(int deep)
     unsigned state = irq_disable();
     __DSB();
     __WFI();
+    /* work around a bug on STM32 systems, see [1] for details, or [2] for
+     * an archive.org backup.
+     * [1]: https://cliffle.com/blog/stm32-wfi-bug
+     * [2]: https://web.archive.org/web/20231205101603/https://cliffle.com/blog/stm32-wfi-bug/
+     */
+#ifdef CPU_STM32
```

Comment:
I have a slight preference for an 

```suggestion
#ifdef CORTEXM_ISB_REQUIRED_AFTER_WFI
```

And enabling that based on `CPU_STM32`. In that case you can move the comment here to the place where it is set for the STM32. It makes it extensible in case other vendors also somehow require this.
It also prevents cluttering the code here with stm32-specifics.

Reply:
Like this?

Reply:
Perfect!